### DISCUSSION TITLE
Pico W with Adafruit LUX sensor VEML7700 - library??

### DISCUSSION THREAD

**Discussion by ESPWarren** [micropython/micropython#15177] [RP2040 / Pico]
Hi

I have a Pico W which is loaded with standard micropython. It is connected via i2c to an Adafruit LUX sensor VEML7700. i2c scan works and detects the opresence of the sensor.

I am struggling to find the appropriate library(ies) to include to make the Pico actually read the sensor.

I am finding several references to an Adafruit CircuitPython solution but I do NOT want to have to use Circuitpython in preference to 'standard' Micropython...

Can anyone tell me which libraries, if any, I can use to read this sensor?

Many thanks

**Comment by ESPWarren** [comment]
OK, someone directly tipped me off with this - it works perfectly on a Pico:

https://github.com/palouf34/veml7700

Note that it seems to require a slower i2c clock - so a freq of 10000 does work well, a freq of 100000 does not - but I am using longer than average cables so do experiment in your situation...

I will leave this for anyone else who follows the trail with the same issue

Thanks

**Reply by scruss** [reply]
Glad you found it. It's listed in the community-maintained [Awesome MicroPython](https://awesome-micropython.com/#light) directory, too.
